### PR TITLE
Feat/refactor-forms-part-VI

### PR DESCRIPTION
**Pull request by matteolc**
### Ticket:

- [23703](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/23703)

### Description

More refactoring on the send-message "wizard". Simplify in view of future re-design and use form action.

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging/components/send-message/SuccessForm.tsx [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
-  const { message, onStep } = context
-
   const t = useTranslations("sendAMessage.SuccessForm")
+  const { message, onStep, errors } = useContext(SendMessageContext)
 
   const partialSuccess =
     (message.userIds?.length || 0) > (message.successfulMessagesCreated || 0)
 
+  console.log(errors?.api)
+
```
Comment (SamSalvatico) [inline apps/messaging/components/send-message/SuccessForm.tsx:14]:
```suggestion

```

File: apps/messaging/components/send-message/ScheduleForm.tsx [hunk 2]
```diff
@@ -4,85 +4,106 @@ import utc from "dayjs/plugin/utc"
...
+          formData.set("templateMetaId", templateMetaId)
+          formData.set("userIds", userIds.join(","))
+          formData.set("subject", subject || "")
+          formData.set("plainText", plainText || "")
+          formData.set("excerpt", excerpt || "")
+          formData.set("transports", transports.join(","))
+          formData.set("schedule", buildSchedule(formData))
+          for (const key of formData.keys()) {
+            console.log(key, formData.get(key))
+          }
```
Comment (SamSalvatico) [inline apps/messaging/components/send-message/ScheduleForm.tsx:103]:
```suggestion
```

File: apps/messaging/components/send-message/SuccessForm.tsx (hunk 1 above)
Comment (matteolc) [reply apps/messaging/components/send-message/SuccessForm.tsx]: https://github.com/ogcio/life-events/commit/f7f71df1e08227ed16634b9ad4b5a6499f970092

File: apps/messaging/components/send-message/ScheduleForm.tsx (hunk 2 above)
Comment (matteolc) [reply apps/messaging/components/send-message/ScheduleForm.tsx]: https://github.com/ogcio/life-events/commit/f7f71df1e08227ed16634b9ad4b5a6499f970092

**Review by SamSalvatico — APPROVED** [review]

**Pull request merged**